MetaMask/metamask-mobile
# fix: 1273 duplicate contact addresses

## **Description**
* Adds a warning icon to contact addresses that are associated with more than 1 chainId (ambiguous)
* Tapping icon opens bottom sheet with info that the address is associated to more than one chain
* When selecting an ambiguous address in the Send To flow, an analytics event is fired

## **Related issues**

_Fixes #[1273](https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/gh/metamask/mobile-planning/1273)

## **E2E Tests**

✅ Pass: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/df69caa4-66f3-44cd-ad41-a097a6491ed7

## **Manual testing steps**

1. Install v7.6 of the app - checkout `release/7.6.0` branch
2. Add Ethereum classic via chainlist.org
3. Add a contact on Mainnet
4. Switch to Ethereum classic
5. Notice that the same added contact shows on ethereum classic chain
6. Checkout this branch. 
7. The app at this point should be able to identify ambiguous addresses and display a warning icon in the contacts list for:
  a. Settings > Contacts
  b. Send To --> Contacts
8. In the send to flow:
  a. If an ambiguous address is selected, you should see a warning message on the Send To screen

## **Screenshots/Recordings**

## Contacts Ambiguous Addresses

https://github.com/MetaMask/metamask-mobile/assets/141057783/e1ae46b1-a952-4422-ab6e-0e55192873ce

### Dark Mode

![Screenshot 2023-10-18 at 10 14 05 AM](https://github.com/MetaMask/metamask-mobile/assets/141057783/c2834584-f833-459c-8770-babe97940609)

## Send To Ambiguous Addresses

https://github.com/MetaMask/metamask-mobile/assets/141057783/8ca774f8-9126-4cf9-9228-2eace070498c

### Dark Mode
![Screenshot 2023-10-18 at 10 14 16 AM](https://github.com/MetaMask/metamask-mobile/assets/141057783/1ae3e97b-40fd-4745-ac9f-115046394cc1)


## Select Ambiguous Addresses to Send To

https://github.com/MetaMask/metamask-mobile/assets/141057783/df97ce68-10ac-4423-a870-b3797e821fa0

### Warning Dark Mode

![Screenshot 2023-10-18 at 10 14 26 AM](https://github.com/MetaMask/metamask-mobile/assets/141057783/73b1421d-da59-47fd-8424-b8fbb5add14d)



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
